packages/loader/container-loader/src/container.ts

```diff
@@ -372,6 +374,29 @@ export interface IPendingDetachedContainerState {
 	hasAttachmentBlobs: boolean;
 }
 
+/**
+ * Ensures only a single instance of the provided async function is running.
+ * If there are multiple calls they will all get the same promise to wait on.
+ */
+const runSingle = <A extends any[], R>(func: (...args: A) => Promise<R>) => {
```

Comment:
Should this be moved to client-utils package instead of being here as it could potentially be used at other places and has nothing specific to do with container.

Reply:
it could. i usually start with thing local, and only move to a common place when i find more usages, but if you think it would be useful, i can put it in a common place to start

Reply:
It is fine here then. I would just put it in a separate file like utils.ts in the same package instead of this file.